app/src/main/java/org/groundplatform/android/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -203,23 +203,24 @@ internal constructor(
   fun getTaskViewModel(taskId: String): AbstractTaskViewModel? {
     val viewModels = taskViewModels.value
 
-    val task = tasks.first { it.id == taskId }
-
     if (viewModels.containsKey(taskId)) {
       return viewModels[taskId]
     }
 
-    return try {
-      val viewModel = viewModelFactory.create(getViewModelClass(task.type))
-      taskViewModels.value[task.id] = viewModel
-
-      val taskData: TaskData? = if (shouldLoadFromDraft) getValueFromDraft(task) else null
-      viewModel.initialize(job, task, taskData)
-      viewModel
-    } catch (e: Exception) {
-      Timber.e("ignoring task with invalid type: $task.type")
-      null
-    }
+    return tasks
+      .firstOrNull { it.id == taskId }
+      ?.let { task ->
+        try {
+          viewModelFactory.create(getViewModelClass(task.type)).apply {
+            taskViewModels.value[task.id] = this
+            val taskData: TaskData? = if (shouldLoadFromDraft) getValueFromDraft(task) else null
+            initialize(job, task, taskData)
+          }
+        } catch (e: Exception) {
+          Timber.e("Ignoring task with invalid type: ${task.type}")
```

Comment:
This reports _any_ Exception as being caused by "invalid type". Is that actually true? Instead, why not report the error with a more generic message, for ex:

`Timber.e(e, "Failed to initialize task view model for task ${task.id}")`